articles/users/search.md

```diff
@@ -0,0 +1,111 @@
+---
+name: User Search
+description: Best practices when searching for users in Auth0
+toc: true
+---
+# User Search
+
+Auth0 provides multiple endpoints you can use to search for users. Each one offers slightly different functionality, and in this document, we'll cover the best practices for using each of the endpoints.
+
+## User Search Options
+
+Currently, Auth0 offers three different ways by which you can search for users:
+
+* [Users](#users)
+* [Users by Email](#users-by-email)
+* [User Export](#user-export)
+
+::: note
+The following sections contain examples on how to call the various user search endpoints. To do so, you'll need to obtain a valid [access token](/api/management/v2/tokens) and provide it in the header of your call (simply replace the `YOUR_MGMT_API_ACCESS_TOKEN` placeholder value).
+:::
+
+### Definitions
+
+In this document, we use the terms **eventually consistent** and **immediately consistent**.
+
+* **Eventually consistent**: When you request information about a user (or a group of users), the response might not reflect the results of a recently-complete write operation. However, if you repeat your request after a short period of time, the response will return up-to-date data.
+
+* **Immediately consistent**: When you request information about a user (or a group of users), the response will reflect the results of all successful write operations, including those that occured shortly prior to your request.
+
+## Users
+
+The [`GET /api/v2/users` endpoint](/api/management/v2#!/Users/get_users) allows you to retrieve a list of users. Using this endpoint, you can:
+
+* Search based on a variety of criteria
+* Select the fields to be returned
+* Sort the returned results
+
+```har
+{
+	"method": "GET",
+	"url": "https://${account.namespace}/api/v2/users",
+	"headers": [{
+		"name": "Authorization",
+		"value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+	}]
+}
+```
+
+This endpoint is **eventually consistent**, and as such, we recommend that you use this endpoint for "back office" processes such as changing the display name of an existing user.
+
+We do **not** recommend that you use this endpoint for:
+
+* Operations that require immediate consistency - please use the [Users by Email endpoint](#users-by-email) for such actions
+* User exports - please use the [User Export endpoint](#user-export) for such actions
+
```

Comment:
Let's add to this list:
Operations that require user search as part of the Authentication Pipeline (use Users by Email)

Reply:
Also add,
Searching for users for Account Linking by email (use Users by Email)